redfish-core/lib/account_service.hpp

```diff
@@ -1997,23 +2132,49 @@ inline void requestAccountServiceRoutes(App& app)
             std::optional<bool> locked;
             std::optional<nlohmann::json> oem;
 
+            if (!json_util::readJson(req, asyncResp->res, "UserName",
```

Comment:
I am giving extra review attention the privilege checking required to perform this operation (above: privileges({{"ConfigureUsers"}, {"ConfigureSelf"}})).
There are two cases:
1. When this method is invoked with an unauthenticated user (because this operation isOnWhitelist()), then "req.session == nullptr" which causes code in http/routing.hpp crow::handle() to call here without performing any privilege checks.  The check for "req.session == nullptr" below handles this case.
2. When this method is invoked with an authenticated user, then req.session is set to a user, and code in the http/routing.hpp crow::handle() checks the required privileges before calling here.
So I think this covers all the the cases correctly.